scarlet/source.py

```diff
@@ -1,6 +1,10 @@
-from .constraint import *
-from .component import *
-from .bbox import *
+from functools import partial
+
+from .constraint import PositivityConstraint, MonotonicityConstraint, SymmetryConstraint
+from .constraint import NormalizationConstraint, ConstraintChain, CenterOnConstraint
+from .parameter import Parameter, relative_step
+from .component import ComponentTree, FunctionComponent, FactorizedComponent
```

Comment:
Is there a reason for this verbosity?

Reply:
It is highly recommended not to use `*` imports in python. At worst it makes code less readable because it is difficult to tell where a given class/method was declared, and at worst can create weird import loops.

Reply:
For example see https://www.geeksforgeeks.org/why-import-star-in-python-is-a-bad-idea/

Reply:
fine